### DISCUSSION TITLE
Communication between servers

### DISCUSSION THREAD

**Discussion by cirospaciari** [uNetworking/uWebSockets#1534] [Q&A]
I opened this discussion to keep these questions out of the issue.

@HagaiVinik comment on the issue https://github.com/uNetworking/uWebSockets/issues/1527 the following:

"I am trying to use the uWS library somehow and make it generate a connection with a other server.
Do you have any suggestions regarding this mechanism?"

**Comment by cirospaciari** [comment]
Personal Tips:

When I need scaling, I used a message broker in this case, Redis pub/sub, another alternative will be RabbitMQ, but Redis pub/sub can do the job very well.

In my case, I need a client because I want to use UNIX domain socket to communicate internally, without any other library. If you cannot use a message broker, I suggest you look in some other native libraries like libwebsockets and do some benchmarks, load tests to see if you can rely on it for client, until uWS get the client working.

libwebsockets supports libuv and I use libuv a lot in my projects this is why that is my temporary alternative to uWS for client. Also I can build it with borringssl with is also my option for SSL when using uWS server.

For sure uWS will perform better than libwebsockets. That's why I will wait and try to contribute when I have more free time. See https://github.com/uNetworking/uWebSockets/discussions/1503

If you want direct communication between 2 servers you need a Client and a Server, uWS provides only a Server for now, so the indication about libwebsockets remains until uWS Clients is released.

**Comment by HagaiVinik** [comment]
@cirospaciari 
Thanks, though the uWS has support for websocket,  can't I start a server and somehow trigger a connect() or send()?
my question is - can I manipulate somehow the library for a simple usage just for triggering connection and sending to a different server, something very basic? not possible at all?

Thanks in advance

**Comment by cirospaciari** [comment]
I did not understand your question, if you want to mock or trigger you can keep the process in a separate function and manually call it.
 If you want to communicate using the WebSocket protocol you need WebSocket a Client and a Server. Normally this client is in a browser like https://developer.mozilla.org/en-US/docs/Web/API/WebSocket but you can use libwebsockets as I recommended before, uWS does not provide a Client yet.
 
 Edit: as recommended before you can also communicate between servers using Redis pub/sub or RabbitMQ.

**Reply by cirospaciari** [reply]
Also, remember you can use app.publish to broadcast, and you can keep track of your connections if you want and use whatever you want, just keep in mind to not use the WebSocket connection after the close trigger
.
